maven-plugin/src/main/java/io/confluent/kafka/schemaregistry/maven/TestLocalCompatibilityMojo.java

```diff
@@ -0,0 +1,154 @@
+      throw new MojoExecutionException(
+          String.format("Invalid schema type %s", schemaType));
+    }
+
+    return schemaProvider.parseSchema(schemaString, references);
+
+  }
+
+  protected ParsedSchema loadSchema(File path, Map<String,
+      SchemaProvider> schemaProviders) throws MojoExecutionException {
+
+    String schemaString;
+    try {
+      schemaString = readFile(path);
+    } catch (IOException e) {
+      throw new MojoExecutionException(
+          String.format("File cannot be found at: %s", path));
+    }
+    List<SchemaReference> references = new ArrayList<>();
+    Optional<ParsedSchema> schema = parseSchema(schemaType, schemaString,
+        references, schemaProviders);
+
+    if (schema.isPresent()) {
+      return schema.get();
+    }
+
+    throw new MojoExecutionException(String.format("Unable to parse schema from %s "
+        + "with schema type as %s", path, schemaType));
+  }
+
+  public void execute() throws MojoExecutionException {
+
+    List<SchemaProvider> providers = defaultSchemaProviders();
+    Map<String, SchemaProvider> schemaProviders = !providers.isEmpty()
+        ? providers.stream().collect(Collectors.toMap(SchemaProvider::schemaType, p -> p))
+        : Collections.singletonMap(AvroSchema.TYPE, new AvroSchemaProvider());
+
+    getLog().debug(String.format("Loading Schema at %s", schemaPath));
+    ParsedSchema schema = loadSchema(schemaPath, schemaProviders);
+
+    getLog().debug("Loading Previous Schemas");
+    ArrayList<ParsedSchema> previousSchemas = new ArrayList<>();
+    for (File previousSchemaPath : previousSchemaPaths) {
+      previousSchemas.add(loadSchema(previousSchemaPath, schemaProviders));
+    }
+
+    CompatibilityChecker checker = CompatibilityChecker.checker(
+        CompatibilityLevel.forName(compatibilityLevel));
+
+    List<String> errorMessages = checker.isCompatible(schema, previousSchemas);
+
+    compatibilityLevel = compatibilityLevel.toUpperCase();
+    if (previousSchemas.size() > 1
+        && (compatibilityLevel.equals("BACKWARD")
+        || compatibilityLevel.equals("FORWARD")
+        || compatibilityLevel.equals("FULL"))) {
+
+      getLog().info(String.format("Checking only with latest Schema at %s",
+          previousSchemaPaths.get(previousSchemaPaths.size() - 1)));
+    }
+
+    success = errorMessages.isEmpty();
+
+    if (success) {
+      getLog().info(String.format("Schema is %s compatible with previous schemas",
+          compatibilityLevel.toLowerCase()));
+    } else {
+      String errorLog = String.format("Schema is not %s compatible with previous schemas %n",
+          compatibilityLevel.toLowerCase()) + errorMessages.get(0);
+      getLog().error(errorLog);
+    }
+
+  }
+
+  private static String readFile(File file) throws IOException {
+    byte[] encoded = Files.readAllBytes(file.toPath());
+    return new String(encoded, StandardCharsets.UTF_8);
+  }
+
+  private List<SchemaProvider> defaultSchemaProviders() {

```

Comment:
Can we add this method to a Utils helper class to avoid duplicate methods?

Reply:
I've added a new class schemaUtils.java which has both the methods. 
SchemaRegistryMojo.java and UploadSchemaRegistryMojo.java use those methods, so I made changes to those files as well.